### PR TITLE
Inline fixes

### PR DESCRIPTION
**Pull request by wilr**

### CODE REVIEW

File: src/GridFieldAddNewInlineButton.php [hunk 1]
```diff
@@ -86,11 +86,11 @@ public function getHTMLFragments($grid)
 
         $fragment = $this->getFragment();
 
-        if (!$editable = $grid->getConfig()->getComponentByType('GridFieldEditableColumns')) {
+        if (!$editable = $grid->getConfig()->getComponentByType(GridFieldEditableColumns::class)) {
             throw new Exception('Inline adding requires the editable columns component');
         }
 
-        Requirements::javascript(THIRDPARTY_DIR . '/javascript-templates/tmpl.js');
+        Requirements::javascript('gridfieldextensions/javascript/tmpl.js');
```
Comment (dhensby) [inline src/GridFieldAddNewInlineButton.php]: As this is 4.0 compat branch you should be using the new resource locator syntax `symbiote/silverstripe-gridfieldextensions:javascript/tmpl.js`

Comment (wilr) [inline src/GridFieldAddNewInlineButton.php]: @dhensby got some documentation on that new resource loader? Didn't see in the docs. Is it just `Requirements::javascript('symbiote/silverstripe-gridfieldextensions:javascript/tmpl.js')` or do you have to use the full example as per the changelog

```
use SilverStripe\View\ThemeResourceLoader;
$moduleFilePath = ModuleLoader::getModule('silverstripe/framework')->getRelativeResourcePath('MyFile.php');
```

Comment (dhensby) [inline src/GridFieldAddNewInlineButton.php]: My example should work see https://github.com/silverstripe/silverstripe-admin/pull/121/files for other usages

**Comment by robbieaverill** [comment]
Can we get this merged? I've ended up repeating some of these fixes in #203 for example. Also, adding the tmpl.js back is nice.

**Comment by robbieaverill** [comment]
This would be nice too, to make the inline add button more bootstrappy:

```patch
diff --git a/templates/Symbiote/GridFieldExtensions/GridFieldAddNewInlineButton.ss b/templates/Symbiote/GridFieldExtensions/GridFieldAddNewInlineButton.ss
index d5d4887..d3bc480 100644
--- a/templates/Symbiote/GridFieldExtensions/GridFieldAddNewInlineButton.ss
+++ b/templates/Symbiote/GridFieldExtensions/GridFieldAddNewInlineButton.ss
@@ -1,3 +1,3 @@
-<button href="$Link" class="ss-gridfield-add-new-inline ss-ui-action-constructive ss-ui-button" data-icon="add">
+<button href="$Link" class="ss-gridfield-add-new-inline btn btn-primary font-icon-plus">
 	$Title
 </button>

```

**Pull request merged**